_posts/2019-11-19-demo-fraud-detection.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: post
+title: "Building a Fraud Detection System with Runtime Rules Updates on Apache Flink"
+date: 2019-11-19T12:00:00.000Z
+authors:
+- alex:
+  name: "Alexander Fedulov"
+  twitter: "afedulov"
+excerpt:
+categories: news
+---
+
+In this series of blog posts you will learn about three powerful Flink patterns for building streaming applications:
+
+ - Dynamic updates of application logic
+ - Dynamic data partitioning (shuffle), controlled at runtime
+ - Low latency alerting based on custom window logic (not using the Window API)
+
+These patterns expand the possibilities of what is achievable with statically defined data flows and thus provide important building blocks to fulfil versatile business requirements.
+
+**Dynamic updates of application logic** allow Flink jobs to change at runtime, without downtime from stopping and resubmitting the code.  
+<br>
+**Dynamic data partitioning** enables the ability to change how events are being distributed and grouped by the `keyBy()` operator in Flink at runtime. Such functionality often becomes a natural requirement when building jobs with dynamically reconfigurable application logic.  
+<br>
+**Custom window management** demonstrates how you can utilize the low level [ProcessFunction API](https://ci.apache.org/projects/flink/flink-docs-stable/dev/stream/operators/process_function.html), when [Window API](https://ci.apache.org/projects/flink/flink-docs-stable/dev/stream/operators/windows.html) is not exactly matching your requirements. Specifically, you will learn how to implement low latency alerting on windows and how to limit state growth with timers.    
+
+The patterns are built up on top of core Flink functionality, however they might not be immediately obvious from the framework's documentation as it is not trivial to explain and give motivation behind them without a concrete example use case. For this reason we will look at the details of the patterns on an example of building an application which represents a common usage scenario for Apache Flink - a _Fraud Detection_ engine.
```

Comment:
```suggestion
The patterns are built on top of core Flink functionality, however, they might not be immediately obvious from the framework's documentation as explaining and presenting the motivation behind those patterns is not always trivial without a concrete use case. This is why we will provide more details about the patterns on a practical example of building an application that represents a common usage scenario for Apache Flink — a _Fraud Detection_ engine.
```

Reply:
1) those patterns of these patterns? I though if we have referenced them before, they are kind of "here" already.
2) "provide more details" - I am a bit on the fence about this one, I think more details means we have provided ANY details before, which we did not. This is the first time they are described actually )

Reply:
I also updated the suggestion based on your comment here. 
Let me know what you think?